include/ignition/math/Vector2.hh

```diff
@@ -108,10 +108,10 @@ namespace ignition
       {
         double d = this->Length();
```

Comment:
The `Length()` function returns a `T` already.  It seems to me like _this_ line is where we should apply a fix (by changing the local variable to be of type `d`), not casting it repeatedly in subsequent lines.